src/components/common/CountDownTimer.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <div />
+</template>
+
+<script>
+import VueTimers from 'vue-timers/mixin';
+
+export default {
+  name: 'CountDownTimer',
+
+  props: {
+    timeout: {
+      type: Number,
+      default: 30000,
+    },
+    isLocked: {
+      type: Boolean,
+      default: false,
+    },
+    counter: {
+      type: Number,
+      default: 0,
+    },
+  },
+
+  watch: {
+    isLocked(newValue) {
+      if (!newValue) {
+        this.stopRequestTimer();
+        return;
+      }
+
+      this.startRequestTimer();
+    },
+  },
+
+  methods: {
+    startRequestTimer() {
+      const initialCounter = this.timeout / 1000;
+
+      this.$emit('update:is-locked', true);
+      this.$emit('update:counter', initialCounter);
+      this.$timer.start('request');
+    },
+
+    stopRequestTimer() {
+      if (!this.timers.request.isRunning) {
+        return;
+      }
+
+      this.$emit('update:is-locked', false);
+      this.$timer.stop('request');
+    },
+
+    onRequestTick() {
+      const nextCounterValue = this.counter - 1;
+
+      this.$emit('update:counter', nextCounterValue);
+
+      if (nextCounterValue === 0) {
+        this.stopRequestTimer();
+      }
+    },
+  },
+
+  mixins: [VueTimers],
+
+  timers: {
+    request: {
```

Comment:
I think it not relates to any request and should has more abstract name

Reply:
Have sense